src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/controller/cas2bail/Cas2BailApplicationController.kt

```diff
@@ -0,0 +1,175 @@
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.jpa.entity.cas2bail.Cas2BailApplicationSummaryEntity
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.problem.BadRequestProblem
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.problem.ConflictProblem
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.problem.ForbiddenProblem
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.problem.NotFoundProblem
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.results.AuthorisableActionResult
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.results.ValidatableActionResult
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.service.HttpAuthService
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.service.NomisUserService
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.service.cas2.OffenderService
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.service.cas2bail.Cas2BailApplicationService
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.transformer.cas2bail.Cas2BailApplicationsTransformer
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.util.PageCriteria
+import java.net.URI
+import java.util.*
+
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.api.model.Cas2ApplicationSummary as ModelCas2ApplicationSummary
+
+@Service(
+  "uk.gov.justice.digital.hmpps.approvedpremisesapi.controller.cas2bail" +
+    ".Cas2BailApplicationsController",
+)
+class Cas2BailApplicationController(
+  private val httpAuthService: HttpAuthService,
+  private val cas2BailApplicationService: Cas2BailApplicationService,
+  private val cas2BailApplicationsTransformer: Cas2BailApplicationsTransformer,
+  private val objectMapper: ObjectMapper,
+  private val offenderService: OffenderService,
+  private val userService: NomisUserService,
+) : ApplicationsCas2bailDelegate {
+
+  override fun applicationsGet(
+    isSubmitted: Boolean?,
+    page: Int?,
+    prisonCode: String?,
+  ): ResponseEntity<List<ModelCas2ApplicationSummary>> {
+    val user = userService.getUserForRequest()
+
+    prisonCode?.let { if (prisonCode != user.activeCaseloadId) throw ForbiddenProblem() }
+
+    val pageCriteria = PageCriteria("createdAt", SortDirection.desc, page)
+
+    val (applications, metadata) = cas2BailApplicationService.getCas2BailApplications(prisonCode, isSubmitted, user, pageCriteria)
+
+    return ResponseEntity.ok().headers(
+      metadata?.toHeaders(),
+    ).body(getPersonNamesAndTransformToSummaries(applications))
+  }
+
+  override fun applicationsApplicationIdGet(applicationId: UUID): ResponseEntity<Application> {
+    val user = userService.getUserForRequest()
+
+    val application = when (
+      val applicationResult = cas2BailApplicationService
+        .getCas2BailApplicationForUser(
+          applicationId,
+          user
+        )
+
+    ) {
+      is AuthorisableActionResult.NotFound -> null
```

Comment:
I think there's an open question as to whether we should be adding more use of deprecate code (in this case using AuthorisableActionResult instead of CasResult). I'm not sure what the answer is, i'll get back to you.

Reply:
my position is we shouldn't use deprecated code, and use the latest pattern we have. please can we do this?